### PR TITLE
Add synonyms for ADO-style connection strings

### PR DESCRIPTION
**Pull request by shueybubbles**
For https://github.com/denisenkom/go-mssqldb/issues/645

I've added mappings for some of the most commonly aliased items - server, user id, and database. I've also added `application name` as a synonym for the `app name` this driver currently supports. There may be other worthwhile synonyms. See <https://docs.microsoft.com/en-us/openspecs/sql_server_protocols/ms-sccstr/775b2bab-a604-4897-87e6-c36ee23dd3f1> for the full list.

I've also reformatted the readme a bit to satisfy the MD linter in vs code.

Queston about github etiquette - should I open this PR against denisenkom/go-mssqldb directly instead? I've not been active in OSS before. 

I am going to queue up a few changes in this fork first and open a PR to push them all over at once, but I can follow other preferred patterns as requested.

### CODE REVIEW

File: conn_str.go [hunk 1]
```diff
@@ -278,6 +289,20 @@ func splitConnectionString(dsn string) (res map[string]string) {
...
 			value = strings.TrimSpace(lst[1])
 		}
+		synonym, hasSynonym := adoSynonyms[name]
+		if hasSynonym {
+			name = synonym
+		}
+		// "server" in ADO can include a protocol and a port.
+		// We only support tcp protocol
+		if name == "server" {
+			value = strings.TrimPrefix(value, "tcp:")
```
Comment (kenvanhyning) [inline conn_str.go:299]: Should we check for other protocols here?  IIUC, passing through unchanged is going to get the generic connection failure and not point specifically to the protocol vs. the host itself being wrong???

Comment (shueybubbles) [reply conn_str.go:299]: I hope the number of people who try to share connection strings from their old ADO apps to their shiny new go app is 1) small, and 2) isolated to people who know the limitations of this driver. I don't see much value in adding defensive code against a named pipes protocol string. I'd expect "Unable to resolve host name np:someserver" is a good enough error at that point.

Comment (kenvanhyning) [reply conn_str.go:299]: ack

**Review by kenvanhyning — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -56,104 +56,114 @@ Other supported formats are listed below.
...
 valid when a new connection is made. Authentication is supported using a callback func that
 provides a fresh and valid token using a connector:
+
 ``` golang
 conn, err := mssql.NewAccessTokenConnector(
   "Server=test.database.windows.net;Database=testdb",
   tokenProvider)
 if err != nil {
-	// handle errors in DSN
+ // handle errors in DSN
```
Comment (ascarter) [inline README.md:128]: Go code uses tabs for indentation.

File: conn_str_test.go [hunk 3]
```diff
@@ -186,10 +201,10 @@ func testConnParams(t testing.TB) connectParams {
 	}
 	if len(os.Getenv("HOST")) > 0 && len(os.Getenv("DATABASE")) > 0 {
 		return connectParams{
-			host: os.Getenv("HOST"),
+			host:     os.Getenv("HOST"),
```
Comment (ascarter) [inline conn_str_test.go:204]: I don't think this func (`testConnParams`) is ever being called. It won't run as a test if it is lowercase and I don't see anything else calling it.

**Review by ascarter — APPROVED** [review]
Looks good to me. Just fix up the readme Go code to look more correct - for better or worse Go uses tabs and those are usually set to 4+ (if not 8)

**Comment by ascarter** [comment]
RE: committing back - the way I usually do it is to set upstream to the original repo and make sure I make feature branches off a sync of the main off the upstream project. Probably squash and clean up your commits into a PR that is concise and doesn't include our internal conversations.

I don't recommend queuing up a lot of changes and just making one big PR. Lots of smaller PR's will be better at least until you really have something worth a big drop. We want the maintainer to build up trust with the PR's we send.

If you reformat whitespace just make a note why you did it so he doesn't decide to focus on petty issues like formatting.

File: conn_str_test.go (hunk 3 above)
Comment (shueybubbles) [reply conn_str_test.go:204]: It's used by other tests in tds_test.go

**Pull request closed without merging**